code/go/0chain.net/smartcontract/storagesc/block_reward_test.go

```diff
@@ -1,286 +1,124 @@
+			sp, err := ssc.getStakePool(bID, balances)
+			require.NoError(t, err)
+			require.EqualValues(t, r.blobberRewards[i], sp.Rewards.Blobber)
+			for j := range p.delegatesBal[i] {
+				key := "delegate" + strconv.Itoa(j)
+				require.EqualValues(t, r.blobberDelegatesRewards[i][j], sp.Pools[key].Rewards)
 			}
-			if reward.serviceChargeCapacity > 0 {
-				balances.On("AddMint", &state.Mint{
-					Minter: ADDRESS, ToClientID: id, Amount: reward.serviceChargeCapacity,
-				}).Return(nil)
-			}
-			if reward.serviceChargeUsage > 0 {
-				balances.On("AddMint", &state.Mint{
-					Minter: ADDRESS, ToClientID: id, Amount: reward.serviceChargeUsage,
-				}).Return(nil)
-			}
-			balances.On("GetTrieNode", stakePoolKey(ssc.ID, id)).Return(&sPool, nil).Once()
 		}
-		balances.On("GetTrieNode", ALL_BLOBBERS_KEY).Return(blobbers, nil).Once()
-
-		for i, sPool := range sPools {
-			i := i
-			sPool := sPool
-			if rewards[i].total == 0 {
-				continue
-			}
-			balances.On(
-				"InsertTrieNode",
-				stakePoolKey(ssc.ID, sPool.Settings.DelegateWallet),
-				mock.MatchedBy(func(sp *stakePool) bool {
-					for key, dPool := range sp.Pools {
-						var wSplit = strings.Split(key, " ")
-						dIndex, err := strconv.Atoi(wSplit[2])
-						require.NoError(t, err)
-						value, ok := sPools[i].Pools[key]
-
-						if !ok ||
-							value.DelegateID != dPool.DelegateID ||
-							dPool.Rewards != rewards[i].delegateUsage[dIndex]+rewards[i].delegateCapacity[dIndex] {
-							return false
-						}
-					}
-					return sp.Rewards.Charge == rewards[i].serviceChargeCapacity+rewards[i].serviceChargeUsage &&
-						sp.Rewards.Blobber == rewards[i].total &&
-						sp.Settings.DelegateWallet == sPool.Settings.DelegateWallet
 
-				}),
-			).Return("", nil).Once()
-		}
-		conf.Minted += zcnToBalance(p.blockReward)
-		balances.On("InsertTrieNode", scConfigKey(ssc.ID), conf).Return("", nil).Once()
-		return ssc, balances
+		_, err := balances.DeleteTrieNode(ACTIVE_PASSED_BLOBBERS_KEY)
+		require.NoError(t, err)
 	}
 
-	type want struct {
-		error    bool
-		errorMsg string
-	}
 	tests := []struct {
-		name       string
-		parameters parameters
-		want       want
+		name    string
+		wantErr bool
+		params  params
+		result  result
 	}{
-
-		{
-			name: "1 blobbers",
-			parameters: parameters{
-				blobberStakes:        [][]float64{{10}},
-				blobberServiceCharge: []float64{0.1},
-				blockReward:          100,
-				qualifyingStake:      10.0,
-				blobberCapacityRatio: 5,
-				blobberUsageRatio:    10,
-				minerRatio:           2,
-				sharderRatio:         3,
-			},
-		},
 		{
-			name: "3 blobbers",
-			parameters: parameters{
-				blobberStakes:        [][]float64{{10}, {5, 5, 10}, {2, 2}},
-				blobberServiceCharge: []float64{0.1, 0.2, 0.3},
-				blockReward:          100,
-				qualifyingStake:      10.0,
-				blobberCapacityRatio: 5,
-				blobberUsageRatio:    10,
-				minerRatio:           2,
-				sharderRatio:         3,
+			name: "1_blobber",
+			params: params{
```

Comment:
Once we have added the calculation parameters to sc.yaml settings, we should have a more realistic test case. With a full partition of blobbers, and add (gamma, a, b, i, k mu) to the `parmas` object. This could allow a user to experiment with different values.